### ISSUE TITLE
Cannot Disable Installer Repos

### ISSUE THREAD

**Issue by tmuntaner** [SUSE/rmt#607]
With the merge of #592, it is impossible to permanently disable an installer update repository. I'm not sure that is the experience we want.

Example:

```
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory?    | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:17:20 UTC |
| 4001   | SLE-Module-Basesystem15-SP2-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:17:56 UTC |
| 3999   | SLE-Module-Basesystem15-SP2-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:18:19 UTC |
| 4058   | SLE-Module-Server-Applications15-SP2-Pool for sle-15-x86_64    | Mandatory     | Mirror  | 2020-07-16 13:18:25 UTC |
| 4056   | SLE-Module-Server-Applications15-SP2-Updates for sle-15-x86_64 | Mandatory     | Mirror  | 2020-07-16 13:18:27 UTC |
| 3963   | SLE-Product-SLES15-SP2-Pool for sle-15-x86_64                  | Mandatory     | Mirror  | 2020-07-16 13:18:29 UTC |
| 3960   | SLE-Product-SLES15-SP2-Updates for sle-15-x86_64               | Mandatory     | Mirror  | 2020-07-16 13:18:28 UTC |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
```

---

```
rmt-cli sync
```

```
I, [2020-08-26T16:28:14.950991 #5424]  INFO -- : Cleaning up the database
I, [2020-08-26T16:28:14.991662 #5424]  INFO -- : Downloading data from SCC
I, [2020-08-26T16:28:14.991736 #5424]  INFO -- : Updating products
I, [2020-08-26T16:30:36.884300 #5424]  INFO -- : Updating repositories
I, [2020-08-26T16:30:38.519684 #5424]  INFO -- : Updating subscriptions
```

---

```
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory?    | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:17:20 UTC |
| 4001   | SLE-Module-Basesystem15-SP2-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:17:56 UTC |
| 3999   | SLE-Module-Basesystem15-SP2-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:18:19 UTC |
| 4058   | SLE-Module-Server-Applications15-SP2-Pool for sle-15-x86_64    | Mandatory     | Mirror  | 2020-07-16 13:18:25 UTC |
| 4056   | SLE-Module-Server-Applications15-SP2-Updates for sle-15-x86_64 | Mandatory     | Mirror  | 2020-07-16 13:18:27 UTC |
| 3963   | SLE-Product-SLES15-SP2-Pool for sle-15-x86_64                  | Mandatory     | Mirror  | 2020-07-16 13:18:29 UTC |
| 3960   | SLE-Product-SLES15-SP2-Updates for sle-15-x86_64               | Mandatory     | Mirror  | 2020-07-16 13:18:28 UTC |
| 2162   | SLE12-SP2-SAP-Installer-Updates for sle-12-x86_64              | Not Mandatory | Mirror  |                         |
| 2706   | SLE15-Installer-Updates for sle-15-x86_64                      | Not Mandatory | Mirror  |                         |
| 3217   | SLE15-SP1-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 3962   | SLE15-SP2-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 4610   | SLE15-SP3-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 2148   | SLED12-SP2-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2215   | SLED12-SP3-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2795   | SLED12-SP4-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2101   | SLES12-SP2-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2191   | SLES12-SP3-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2771   | SLES12-SP4-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 3650   | SLES12-SP5-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
```

---

```
rmt-cli repos disable 3650
```

```
Repository by ID 3650 successfully disabled.

To clean up downloaded files, please run 'rmt-cli repos clean'
```

---

```
rmt-cli sync
```

```
I, [2020-08-26T16:28:14.950991 #5424]  INFO -- : Cleaning up the database
I, [2020-08-26T16:28:14.991662 #5424]  INFO -- : Downloading data from SCC
I, [2020-08-26T16:28:14.991736 #5424]  INFO -- : Updating products
I, [2020-08-26T16:30:36.884300 #5424]  INFO -- : Updating repositories
I, [2020-08-26T16:30:38.519684 #5424]  INFO -- : Updating subscriptions
```

---

```
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory?    | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:17:20 UTC |
| 4001   | SLE-Module-Basesystem15-SP2-Pool for sle-15-x86_64             | Mandatory     | Mirror  | 2020-07-16 13:17:56 UTC |
| 3999   | SLE-Module-Basesystem15-SP2-Updates for sle-15-x86_64          | Mandatory     | Mirror  | 2020-07-16 13:18:19 UTC |
| 4058   | SLE-Module-Server-Applications15-SP2-Pool for sle-15-x86_64    | Mandatory     | Mirror  | 2020-07-16 13:18:25 UTC |
| 4056   | SLE-Module-Server-Applications15-SP2-Updates for sle-15-x86_64 | Mandatory     | Mirror  | 2020-07-16 13:18:27 UTC |
| 3963   | SLE-Product-SLES15-SP2-Pool for sle-15-x86_64                  | Mandatory     | Mirror  | 2020-07-16 13:18:29 UTC |
| 3960   | SLE-Product-SLES15-SP2-Updates for sle-15-x86_64               | Mandatory     | Mirror  | 2020-07-16 13:18:28 UTC |
| 2162   | SLE12-SP2-SAP-Installer-Updates for sle-12-x86_64              | Not Mandatory | Mirror  |                         |
| 2706   | SLE15-Installer-Updates for sle-15-x86_64                      | Not Mandatory | Mirror  |                         |
| 3217   | SLE15-SP1-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 3962   | SLE15-SP2-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 4610   | SLE15-SP3-Installer-Updates for sle-15-x86_64                  | Not Mandatory | Mirror  |                         |
| 2148   | SLED12-SP2-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2215   | SLED12-SP3-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2795   | SLED12-SP4-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2101   | SLES12-SP2-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2191   | SLES12-SP3-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 2771   | SLES12-SP4-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
| 3650   | SLES12-SP5-Installer-Updates for sle-12-x86_64                 | Not Mandatory | Mirror  |                         |
+--------+----------------------------------------------------------------+---------------+---------+-------------------------+
```

Repo `3650` is back!!!!!!

**Comment by tmuntaner** [comment]
@jesusbv Was this an intention of PR #592?

If so, I think we need to make this experience more clear in the changelog. "By default" doesn't read as permanent to me.

**Comment by tmuntaner** [comment]
I talked to @jesusbv and the requirements were not well defined in #554. We will better define the requirements tomorrow and introduce a fix.

**Comment by tmuntaner** [comment]
To me, @jesusbv, I think your initial [code](https://github.com/SUSE/rmt/commit/31346f86c301b4b18ae16864322eeda3144f33a5#diff-7b1085760b33aef4c8356bfa224dafdfR174) was on the right path.

It seems that there was some initial confusion that came in with https://github.com/SUSE/rmt/pull/592#issuecomment-668691925 by @digitaltom. Normally, that comment would be correct, as it's the column to enable repository mirroring. The problem is that it's the code path to sync with SCC, not enable/disable a repository.

If we set the installer repo to `enabled = true` and not `mirroring_enabled = true`, `rmt-cli products enable FOO` can then enable the repository for us. Otherwise, we'll enable every installer repository that exists and it will be next to impossible to disable them.

Previous behavior for enable:

```
rmt-cli products enable sles/15.1/x86_64
```

```
Found product by target sles/15.1/x86_64: SUSE Linux Enterprise Server 15 SP1 x86_64.
Enabling SUSE Linux Enterprise Server 15 SP1 x86_64:
  SUSE Linux Enterprise Server 15 SP1 x86_64:
    Enabled repository SLE-Product-SLES15-SP1-Pool.
    Enabled repository SLE-Product-SLES15-SP1-Updates.
  Basesystem Module 15 SP1 x86_64:
    Enabled repository SLE-Module-Basesystem15-SP1-Pool.
    Enabled repository SLE-Module-Basesystem15-SP1-Updates.
  Server Applications Module 15 SP1 x86_64:
    Enabled repository SLE-Module-Server-Applications15-SP1-Pool.
    Enabled repository SLE-Module-Server-Applications15-SP1-Updates.
```

```
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory? | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory  | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory  | Mirror  | 2020-07-16 13:17:20 UTC |
| 3303   | SLE-Module-Server-Applications15-SP1-Pool for sle-15-x86_64    | Mandatory  | Mirror  |                         |
| 3301   | SLE-Module-Server-Applications15-SP1-Updates for sle-15-x86_64 | Mandatory  | Mirror  |                         |
| 3218   | SLE-Product-SLES15-SP1-Pool for sle-15-x86_64                  | Mandatory  | Mirror  |                         |
| 3215   | SLE-Product-SLES15-SP1-Updates for sle-15-x86_64               | Mandatory  | Mirror  |                         |
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
```

The installer repo is missing.

---

With the change `repo_item[:enabled] = true if repo_item[:installer_updates]` from your previous PR.

```
rmt-cli sync
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory? | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory  | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory  | Mirror  | 2020-07-16 13:17:20 UTC |
| 3303   | SLE-Module-Server-Applications15-SP1-Pool for sle-15-x86_64    | Mandatory  | Mirror  |                         |
| 3301   | SLE-Module-Server-Applications15-SP1-Updates for sle-15-x86_64 | Mandatory  | Mirror  |                         |
| 3218   | SLE-Product-SLES15-SP1-Pool for sle-15-x86_64                  | Mandatory  | Mirror  |                         |
| 3215   | SLE-Product-SLES15-SP1-Updates for sle-15-x86_64               | Mandatory  | Mirror  |                         |
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
```

```
rmt-cli products enable sles/15.1/x86_64
```

```
Found product by target sles/15.1/x86_64: SUSE Linux Enterprise Server 15 SP1 x86_64.
Enabling SUSE Linux Enterprise Server 15 SP1 x86_64:
  SUSE Linux Enterprise Server 15 SP1 x86_64:
    Enabled repository SLE15-SP1-Installer-Updates.
  Basesystem Module 15 SP1 x86_64:
    All repositories have already been enabled.
  Server Applications Module 15 SP1 x86_64:
    All repositories have already been enabled.
```

```
rmt-cli repos ls
```

```
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| SCC ID | Product                                                        | Mandatory? | Mirror? | Last mirrored           |
+--------+----------------------------------------------------------------+------------+---------+-------------------------+
| 3263   | SLE-Module-Basesystem15-SP1-Pool for sle-15-x86_64             | Mandatory  | Mirror  | 2020-07-16 13:14:49 UTC |
| 3261   | SLE-Module-Basesystem15-SP1-Updates for sle-15-x86_64          | Mandatory  | Mirror  | 2020-07-16 13:17:20 UTC |
| 3303   | SLE-Module-Server-Applications15-SP1-Pool for sle-15-x86_64    | Mandatory  | Mirror  |                         |
| 3301   | SLE-Module-Server-Applications15-SP1-Updates for sle-15-x86_64 | Mandatory  | Mirror  |                         |
| 3218   | SLE-Product-SLES15-SP1-Pool for sle-15-x86_64                  | Mandatory  | Mirror  |                         |
| 3215   | SLE-Product-SLES15-SP1-Updates for sle-15-x86_64               | Mandatory  | Mirror  |                         |
| 3217   | SLE15-SP1-Installer-Updates for sle-15-x86_64                  | Mandatory  | Mirror  |                         |
+--------+----------------------------------------------------------------+------------+---------+-------------------------
```

The installer repo is there and that is exactly the behavior that I would expect.

@digitaltom Would you agree with this assessment?

**Comment by digitaltom** [comment]
I agree, it's unexpected that the repo comes back automatically after disabling it. It was just meant to be there automatically when enabling the product.

**Comment by jesusbv** [comment]
Yes, that was my intention with the original approach. Thank you @tmuntaner for clarifying. 

I will have a new PR addressing the fix.

**Comment by tmuntaner** [comment]
@jesusbv do you need any help with this PR? The RMT integration tests are failing because of this issue.

**Comment by tmuntaner** [comment]
We had to merge the integration tests fixes of #608, when we resolve this issue, we can revert the change.

**Comment by jesusbv** [comment]
No, no problem with the PR yet, I will try to have it ASAP.
